wp-content/civi-extensions/goonjcustom/Civi/GoonjActivitiesService.php

```diff
@@ -0,0 +1,539 @@
+      return;
+    }
+
+    if (!($contactId = self::findGoonjActivitiesInitiatorContact($params))) {
+      return;
+    }
+
+    $collectionCampId = $contactId['entity_id'];
+
+    $collectionCamp = EckEntity::get('Collection_Camp', FALSE)
+      ->addSelect('Collection_Camp_Core_Details.Contact_Id', 'custom.*')
+      ->addWhere('id', '=', $collectionCampId)
+      ->execute()->single();
+
+    $contactId = $collectionCamp['Collection_Camp_Core_Details.Contact_Id'];
+
+    $optionValue = OptionValue::get(FALSE)
+      ->addWhere('option_group_id:name', '=', 'activity_type')
+      ->addWhere('label', '=', 'Induction')
+      ->execute()->single();
+
+    $activityTypeId = $optionValue['value'];
+
+    $induction = Activity::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('target_contact_id', '=', $contactId)
+      ->addWhere('activity_type_id', '=', $activityTypeId)
+      ->addOrderBy('created_date', 'DESC')
+      ->setLimit(1)
+      ->execute()->single();
+
+    $inductionId = $induction['id'];
+
+    EckEntity::update('Collection_Camp', FALSE)
+      ->addValue('Collection_Camp_Intent_Details.Initiator_Induction_Id', $inductionId)
+      ->addWhere('id', '=', $collectionCampId)
+      ->execute();
+  }
+
+  /**
+   * This hook is called after the database write on a custom table.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The custom group ID.
+   * @param int $objectId
+   *   The entityID of the row in the custom table.
+   * @param object $objectRef
+   *   The parameters that were sent into the calling function.
+   */
+  public static function setOfficeDetails($op, $groupID, $entityID, &$params) {
+    if ($op !== 'create' || self::getEntitySubtypeName($entityID) !== self::ENTITY_SUBTYPE_NAME) {
+      return;
+    }
+
+    if (!($stateField = self::findStateField($params))) {
+      return;
+    }
+
+    $stateId = $stateField['value'];
+    $goonjActivitiesId = $stateField['entity_id'];
+
+    if (!$stateId) {
+      \CRM_Core_Error::debug_log_message('Cannot assign Goonj Office to goonj activities: ' . $goonjActivitiesId);
+      \CRM_Core_Error::debug_log_message('No state provided on the intent for goonj activities: ' . $goonjActivitiesId);
+      return FALSE;
+    }
+
+    $officesFound = Contact::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('contact_type', '=', 'Organization')
+      ->addWhere('contact_sub_type', 'CONTAINS', 'Goonj_Office')
+      ->addWhere('Goonj_Office_Details.Collection_Camp_Catchment', 'CONTAINS', $stateId)
+      ->execute();
+
+    $stateOffice = $officesFound->first();
+
+    // If no state office is found, assign the fallback state office.
+    if (!$stateOffice) {
+      $stateOffice = self::getFallbackOffice();
+    }
+
+    $stateOfficeId = $stateOffice['id'];
+
+    $rs = EckEntity::update('Collection_Camp', FALSE)
```

Comment:
Please use better variable naming and i don't think we need variable here as it's an update function